### DISCUSSION TITLE
Why isn't my checker detected?

### DISCUSSION THREAD

**Discussion by JopdeBoo** [colour-science/colour-checker-detection#44] [Q&A]
Hi,
The provided example code works well with the example pictures, but i cannot get it to work with an image I took myself. No swatches at all are detected. Can someone help me fix this problem?

![Screenshot 2022-02-28 172330](https://user-images.githubusercontent.com/57945462/156019529-df29f36b-ac28-4a4e-b757-1d24e53a12db.jpg)
![Screenshot 2022-02-28 172419](https://user-images.githubusercontent.com/57945462/156019520-0dc17a01-435e-4595-9850-e2502157ce9b.jpg)

**Comment by KelSolaar** [comment]
Hi @JopdeBoo,

I highly suspect that this checker is not segment properly, can you try to display the segmented image like in the example: https://nbviewer.org/github/colour-science/colour-checker-detection/blob/develop/colour_checker_detection/examples/examples_detection.ipynb#Additional-Data-Plotting

Cheers,

Thomas

**Comment by JopdeBoo** [comment]
You are right. The checker is not segmented properly: 

![Screenshot 2022-03-01 110548](https://user-images.githubusercontent.com/57945462/156149082-3eb83ca7-16e1-4e93-8197-2f3a6e90ab80.jpg)![Screenshot 2022-03-01 110552](https://user-images.githubusercontent.com/57945462/156149067-04eea6bb-45ef-4500-88a4-24efc0637e4f.jpg)

I have tried cropping the image to get a better result, but this does not seem to work either.

![Screenshot 2022-03-01 111404](https://user-images.githubusercontent.com/57945462/156150967-e4bbc37b-df31-4bce-8600-ba3cdea143c8.jpg)

Are there more settings I need to change to get a better result? The following images are the images I originally took to perform the script on. 

![real_source3](https://user-images.githubusercontent.com/57945462/156150489-b96152d4-a5f2-4fca-84d0-a76b2dfc81d2.jpg)
![real_source2](https://user-images.githubusercontent.com/57945462/156150494-feecbdac-6468-4043-82e9-88054f68ce49.jpg)

Do you have an idea on how to get the segmentation to work well for my case?

**Reply by KelSolaar** [reply]
The process tries to find well defined squares in the image. Here, in your case and because the edges of the samples are very jagged, it is not finding enough of them. Would you be able to clean them in any way? Producing a well cut mask with some black paper that you layer on top would maybe help I think.

**Reply by KelSolaar** [reply]
You could even go down the road of laser cutting the mask, randomly found online link: https://www.ponoko.com/laser-cutting
